File: fortune-blocking-mariadb/src/main/java/me/escoffier/fortune/mariadb/FortuneResource.java
package me.escoffier.fortune.mariadb;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;

import java.util.List;

@Path("/fortunes")
public class FortuneResource {

    private final FortuneRepository repository;

    public FortuneResource(FortuneRepository repository) {
        this.repository = repository;
    }

    @GET
    public List<Fortune> getAllFortunes() {
        return this.repository.listAll();
    }

    @GET
    @Path("/random")
    public Fortune getRandomFortune() {
        return this.repository.getRandomFortune();
    }

}

File: fortune-kotlin-mariadb/src/main/kotlin/me/escoffier/fortune/kotlin/mariadb/FortuneResource.kt
package me.escoffier.fortune.kotlin.mariadb

import io.smallrye.mutiny.coroutines.awaitSuspending
import jakarta.enterprise.inject.Default
import jakarta.inject.Inject
import jakarta.ws.rs.GET
import jakarta.ws.rs.Path

@Path("/fortunes")
class FortuneResource {

    @Inject
    @field: Default
    lateinit var repository: FortuneRepository;


    @GET
    suspend fun getAllFortunes(): List<Fortune> {
        return repository.listAll().awaitSuspending();
    }

    @GET
    @Path("/random")
    suspend fun getRandomFortune(): Fortune {
        return repository.randomFortune.awaitSuspending();
    }
}